packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue

```diff
@@ -21,49 +21,80 @@
-            :to="n.path"
-            :appearance="n.active ? 'raw-inverse' : 'raw'"
-            :variation="n.active ? 'primary' : 'passive'"
-            :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
-          >
-            <span class="icon-box">
-              <oc-icon :name="n.icon" variation="inherit" />
-            </span>
-            <span v-text="$gettext(n.title)" />
-            <oc-icon v-if="n.active" name="check" class="active-check" variation="inherit" />
-          </oc-button>
-        </li>
-      </oc-list>
+      <div style="display: block; position: relative">
+        <oc-list class="applications-list">
+          <li v-for="(n, nid) in applicationsList" :key="`apps-menu-${nid}`" @click="clickApp(n)">
+            <oc-button
+              :key="n.url ? 'apps-menu-external-link' : 'apps-menu-internal-link'"
+              :type="n.url ? 'a' : 'router-link'"
+              :target="n.target"
+              :href="n.url"
+              :to="n.path"
+              :appearance="n.active ? 'raw-inverse' : 'raw'"
+              :variation="n.active ? 'primary' : 'passive'"
+              :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
+            >
+              <oc-application-icon :icon="n.icon" :color-primary="n.color" ref="appIcons" />
+              <span v-text="$gettext(n.title)" />
+            </oc-button>
+          </li>
+        </oc-list>
+      </div>
     </oc-drop>
   </nav>
 </template>
 
 <script lang="ts">
-import { defineComponent, PropType, ComponentPublicInstance } from 'vue'
+import { defineComponent, PropType, ComponentPublicInstance, ref, unref, computed } from 'vue'
 import { configurationManager } from 'web-pkg/src/configuration'
 import { urlJoin } from 'web-client/src/utils'
 import { OcDrop } from 'design-system/src/components'
+import OcApplicationIcon from 'design-system/src/components/OcApplicationIcon/OcApplicationIcon.vue'
+import { useClientService } from 'web-pkg/src'
+import { useGettext } from 'vue3-gettext'
 
 export default defineComponent({
+  components: {
+    OcApplicationIcon
+  },
   props: {
     applicationsList: {
       type: Array as PropType<any[]>,
       required: false,
       default: () => []
     }
   },
-  computed: {
-    applicationSwitcherLabel() {
-      return this.$gettext('Application Switcher')
+  setup() {
+    const appIcons = ref([])
+    const clientService = useClientService()
+    const { $gettext } = useGettext()
+
+    const applicationSwitcherLabel = computed(() => {
+      return $gettext('Application Switcher')
+    })
+    const updateAppIcons = () => {
+      unref(appIcons).forEach((appIcon) => {
+        unref(appIcon).update()
+      })
+    }
+    const setClassicUIDefault = () => {
+      const url = urlJoin(configurationManager.serverUrl, '/index.php/apps/web/settings/default')
+      return clientService.httpAuthenticated.post(url, { isDefault: false })
+    }
```

Comment:
While you're at it, the classic app stuff can be removed as we dropped oC10 support.